tools/tsp-client/src/options.ts

```diff
@@ -50,35 +64,53 @@ export async function getOptions(): Promise<Options> {
     process.exit(0);
   }
 
-  if (!values.emitter) {
-    Logger.error("Option --emitter/-e is required");
-    printUsage();
-    process.exit(1);
+  if (values.emitter) {
+    let emitter = values.emitter.toLowerCase();
+    if (emitter in languageAliases) {
+      emitter = languageAliases[emitter]!;
+    }
+    if (!(knownLanguages as readonly string[]).includes(emitter)) {
+      Logger.error(`Unknown language ${values.emitter}`);
+      Logger.error(`Valid languages are: ${knownLanguages.join(", ")}`);
+      printUsage();
+      process.exit(1);
+    }
   }
 
-  let emitter = values.emitter.toLowerCase();
-  if (emitter in languageAliases) {
-    emitter = languageAliases[emitter]!;
-  }
-  if (!(knownLanguages as readonly string[]).includes(emitter)) {
-    Logger.error(`Unknown language ${values.emitter}`);
-    Logger.error(`Valid languages are: ${knownLanguages.join(", ")}`);
+  if (positionals.length === 0) {
+    Logger.error("Command is required");
     printUsage();
     process.exit(1);
   }
 
-  if (positionals.length === 0 || !positionals[0]) {
-    Logger.error("Output directory is required");
+  if (positionals[0] !== "sync" && positionals[0] !== "generate" && positionals[0] !== "update" && positionals[0] !== "init") {
+    Logger.error(`Unknown command ${positionals[0]}`);
     printUsage();
     process.exit(1);
   }
-  const outputDir = path.resolve(path.normalize(positionals[0]));
+
+  if (positionals[0] === "init") {
+    if (!values.tspConfig) {
+      Logger.error("tspConfig is required");
+      printUsage();
+      process.exit(1);
+    }
+  }
+  // By default, assume that the command is run from the output directory
```

Comment:
Assuming the cwd is the output directory seems.... not super reliable if someone is trying out the command from the repo root for example. Is there a reason we can't have this be explicit?

Reply:
We want to reduce the number of inputs that a service partner needs in order to use this tool. For service partners working from their project directory it makes it simpler, but that is why we still allow them to use the output directory if they're not in the service project directory

Reply:
yeah I'm more thinking, can we perhaps have a prompt before execution like `generating in <cwd here>, continue? [Y/n]` to prevent careless pollution of the tree.

Alternatively, I was thinking it would be somewhat trivial to check the path and look to see if we're in `<something>azure-sdk-for-<thing>/sdk/<someservice>/<somepackage>` and warn/error if not given an explicit path

I'll leave it up to you what seems most useful